File: SkyEditor.Core/ConsoleCommands/ListPlugins.vb
Imports System.Reflection
Namespace ConsoleCommands
    Public Class ListPlugins
        Inherits ConsoleCommand

        Public Overrides Sub Main(arguments() As String)
            Console.WriteLine("Plugins:")
            For Each item In CurrentPluginManager.Plugins
                Console.WriteLine($"{item.PluginName} ({item.GetType.GetTypeInfo.Assembly.FullName})")
            Next
        End Sub
    End Class
End Namespace

File: SkyEditor.Core/ConsoleCommands/Commands/ListPlugins.cs
using SkyEditor.Core.IO;
using System;
using System.Collections.Generic;
using System.Reflection;
using System.Text;

namespace SkyEditor.Core.ConsoleCommands.Commands
{
    public class ListPlugins : ConsoleCommand
    {
        public ListPlugins(PluginManager pluginManager)
        {
            CurrentPluginManager = pluginManager;
        }

        protected PluginManager CurrentPluginManager { get; }

        protected override void Main(string[] arguments)
        {
            Console.WriteLine("Plugins:");
            foreach (var item in CurrentPluginManager.GetPlugins())
            {
                Console.WriteLine($"{item.PluginName} ({item.GetType().GetTypeInfo().Assembly.FullName})");
            }
        }
    }
}
